packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -178,13 +226,52 @@ def setup_vip_workload_tests(dcos_api_session, container, vip_net, proxy_net):
     # a fresh state, but in this case the previous app load may still be winding down,
     # so allow a larger buffer time
     retry_on_result=lambda res: res is None)
-def wait_for_tasks_healthy(dcos_api_session, app_definition):
-    proxy_info = dcos_api_session.marathon.get('v2/apps/{}'.format(app_definition['id'])).json()
-    if proxy_info['app']['tasksHealthy'] == app_definition['instances']:
-        return proxy_info
+def wait_for_tasks_healthy(dcos_api_session, pod, app_definition):
+    if pod:
+        proxy_info = dcos_api_session.marathon.get('v2/pods/{}::status'.format(app_definition['id'])).json()
+        if 'status' in proxy_info and proxy_info['status'] == 'STABLE':
+            return proxy_info
+    else:
+        proxy_info = dcos_api_session.marathon.get('v2/apps/{}'.format(app_definition['id'])).json()
+        if proxy_info['app']['tasksHealthy'] == app_definition['instances']:
+            return proxy_info
     return None
 
 
+@pytest.mark.slow
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
+@pytest.mark.parametrize('container,vip_net,proxy_net', generate_vip_app_permutations())
+def test_pod_vip(dcos_api_session, container: Container, vip_net: Network, proxy_net: Network):
+    ''' Same as test_vip but for pods
```

Comment:
[Minor]: There is a format for doc-string that could be adopted here: https://github.com/dcos/dcos/blob/master/docs/styleguides/python.md

* It will have triple double-quoted strings.
* Args, Returns, and Raises.